components/brave_wallet/browser/BUILD.gn

```diff
@@ -388,6 +388,7 @@ source_set("hd_keyring") {
     "internal:hd_key",
     "//base",
     "//brave/components/brave_wallet/common",
+    "//brave/components/brave_wallet/common:buildflags",
```

Comment:
Why is this added here? I don't see any files in this target that use it. Most likely this was an incorrect fix for making this a public dep in `internal/BUILD.gn`

Reply:
> Why is this added here? I don't see any files in this target that use it. 

zcash_keyring.cc uses the buildflag

> Most likely this was an incorrect fix for making this a public dep in internal/BUILD.gn

Why do you think buildflags should be public deps in internal?

Reply:
they sometimes need to be public deps when they are used in public header files because otherwise you can end up with errors where the header file can't be found. The `buildflag_header` target itself has a public dep on `//build:buildflag_header_h`, but public_deps only go up one level